### PR TITLE
enable more trt options

### PR DESCRIPTION
**Pull request by askhade**
Enabling following options for TRT EP

- ORT_TENSORRT_INT8_ENABLE
- ORT_TENSORRT_INT8_CALIBRATION_TABLE_NAME
- ORT_TENSORRT_INT8_USE_NATIVE_CALIBRATION_TABLE
- ORT_TENSORRT_ENGINE_CACHE_ENABLE
- ORT_TENSORRT_CACHE_PATH

### CODE REVIEW

File: src/onnxruntime.cc [hunk 1]
```diff
@@ -278,6 +283,28 @@ ModelState::LoadModel(
                           value_string, &max_workspace_size_bytes));
                       trt_options.trt_max_workspace_size =
                           static_cast<size_t>(max_workspace_size_bytes);
+                    } else if (param_key == "int8_calibration_table_name") {
+                      RETURN_IF_ERROR(params.MemberAsString(
+                          param_key.c_str(), &int8_calibration_table_name));
+                      trt_options.trt_int8_calibration_table_name = &int8_calibration_table_name;
```
Comment (efeguney) [inline src/onnxruntime.cc:289]:
```suggestion
                      trt_options.trt_int8_calibration_table_name = int8_calibration_table_name.c_str();
```
It looks like trt_int8_calibration_table_name  is cont char*, and assigning string* to char* doesn't seem right.

File: src/onnxruntime.cc [hunk 2]
```diff
@@ -278,6 +283,28 @@ ModelState::LoadModel(
...
+                      RETURN_IF_ERROR(params.MemberAsString(
+                          param_key.c_str(), &value_string));
+                      bool enable_cache;
+                      RETURN_IF_ERROR(ParseBoolValue(
+                          value_string, &enable_cache));
+                      trt_options.trt_engine_cache_enable = enable_cache;
+                    } else if (param_key == "trt_engine_cache_path") {
+                      RETURN_IF_ERROR(params.MemberAsString(
+                          param_key.c_str(), &trt_engine_cache_path));
+                      trt_options.engine_cache_path = &trt_engine_cache_path;
```
Comment (efeguney) [inline src/onnxruntime.cc:307]:
```suggestion
                      trt_options.engine_cache_path = trt_engine_cache_path.c_str();
```
Same as above.

File: src/onnxruntime.cc [hunk 3]
```diff
@@ -264,6 +266,9 @@ ModelState::LoadModel(
                           param_key.c_str(), &value_string));
                       if (value_string == "FP16") {
                         trt_options.trt_fp16_enable = 1;
+                      } else if (value_string == "INT8") {
+                        trt_options.trt_int8_enable = 1;
+                        trt_options.trt_int8_calibration_table_name = &int8_calibration_table_name;
```
Comment (efeguney) [inline src/onnxruntime.cc:271]:
```suggestion
                        trt_options.trt_int8_calibration_table_name = int8_calibration_table_name.c_str();
```
trt_int8_calibration_table_name seems to be const char* based on the definition of OrtTensorRTProviderOptions

**Review by efeguney — CHANGES_REQUESTED** [review]
Thank you for the PR. It looks like we may need to change the assignments to const char* members of OrtTensorRTProviderOptions.

File: src/onnxruntime.cc (hunk 3 above)
Comment (deadeyegoodwin) [reply src/onnxruntime.cc:271]: I'm surprised that the compile didn't fail due to the type mismatch

Comment (askhade) [reply src/onnxruntime.cc:271]: I did not pick the changes correctly ... I had staged this file already and then added this fix after seeing compilation error... but forgot to stage this fix before the commit...

**Review by deadeyegoodwin — APPROVED** [review]

**Pull request merged**